src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php

```diff
@@ -84,13 +86,23 @@ public function configureOptions(OptionsResolver $resolver)
 
         $resolver->setAllowedTypes('html5', 'bool');
 
+        $resolver->setAllowedValues('model_type', ['float', 'integer']);
+
         $resolver->setNormalizer('grouping', static function (Options $options, $value) {
             if ($value && $options['html5']) {
                 throw new LogicException('Cannot use the "grouping" option when the "html5" option is enabled.');
             }
 
             return $value;
         });
+
+        $resolver->setNormalizer('model_type', static function (Options $options, $value) {
+            if ('integer' === $value && 1 === $options['divisor']) {
+                throw new LogicException('Can use integer "model_type" option only when the "divisor" option is not 1.');
```

Comment:
```suggestion
                throw new LogicException('When the "model_type" option is set to "integer", the "divisor" option should not be set to "1".');
```

but why?

Reply:
Because if you set the divisor to 1, you can end up with a float instead of an integer, and lose decimals in the casting

Reply:
I'm sorry I don't get it. Also the original PR had exactly the opposite : throwing when the divisor is *not* 1.
https://github.com/symfony/symfony/pull/50720/files#diff-eb30dc48f2b749fa5d76ce9598bb51c00fcb97bd7a10337f0630db00f5ba3b9aR100

I'm pretty sure I'm missing something :grimacing:

Reply:
I know, and indeed I added a note about that when I opened this PR.